cpp/order/Steinhardt.cc

```diff
@@ -90,15 +101,35 @@ void Steinhardt::compute(const freud::locality::NeighborList* nlist,
 void Steinhardt::baseCompute(const freud::locality::NeighborList* nlist,
                              const freud::locality::NeighborQuery* points, freud::locality::QueryArgs qargs)
 {
-    const auto normalizationfactor = float(4.0 * M_PI / m_num_ms);
+    std::vector<float> normalizationfactor(m_ls.size());
+    for (size_t l_index = 0; l_index < m_ls.size(); ++l_index)
+    {
+        normalizationfactor[l_index] = float(4.0 * M_PI / m_num_ms[l_index]);
+    }
     // For consistency, this reset is done here regardless of whether the array
     // is populated in baseCompute or computeAve.
-    m_qlm_local.reset();
+    for (auto& qlm_local : m_qlm_local)
+    {
+        qlm_local.reset();
+    }
+
     freud::locality::loopOverNeighborsIterator(
         points, points->getPoints(), m_Np, qargs, nlist,
         [=](size_t i, const std::shared_ptr<freud::locality::NeighborPerPointIterator>& ppiter) {
             float total_weight(0);
             const vec3<float> ref((*points)[i]);
+            // Construct PointSPHEvaluator outside loop since the construction is costly.
+            auto max_l = *std::max_element(m_ls.begin(), m_ls.end());
+            fsph::PointSPHEvaluator<float> sph_eval(max_l);
```

Comment:
I could be wrong, but since the `PointSPHEvaluator` precomputes quantities based on theta and phi we wouldn't want to capture by reference since this would lead to race conditions when `compute` is called with the current thread's theta and phi.

Reply:
You're right, ignore this suggestion.